### PR TITLE
Initial Checkout api implementation.  

### PR DESCRIPTION
**Pull request by Sandvoxel**

### CODE REVIEW

**Comment by Sandvoxel** [comment]
I have not tested this api yet.Sometime between the end of the week I am going to test the order request as I plan to use it with my website.

File: squareup/src/api/catalog_api.rs [hunk 1]
```diff
@@ -157,7 +157,7 @@ impl CatalogApi {
         &self,
         params: &ListCatalogParameters,
     ) -> Result<ListCatalogResponse, SquareApiError> {
-        let url = format!("{}/list{}", &self.url(), params.to_query_string()).replace("\"", "");
+        let url = format!("{}/list{}", &self.url(), params.to_query_string());
```
Comment (Sandvoxel) [inline squareup/src/api/catalog_api.rs]: Without this replace you the api call fails because it buts an extra set of quotes around the prams.

Comment (kcable194) [reply squareup/src/api/catalog_api.rs]: Yeah I'm putting it in the to_query_string() function to make the api function a little cleaner. I was writing a custom deserializer but ehhh, I don't want to maintain that

Comment (Sandvoxel) [reply squareup/src/api/catalog_api.rs]: Sounds good.

Comment (kcable194) [reply squareup/src/api/catalog_api.rs]: I'll be pretty busy this weekend but if you want me to keep working on the checkout api, I will get back to it next weekend. Idk what your plans are but good work!

Comment (Sandvoxel) [reply squareup/src/api/catalog_api.rs]: Ive just been working with my local branch I will probably get to implementing the checkout api on my website this weekend or sometime next week. If I run into any issues Ill just report them here as my fork has diverged.

**Comment by Sandvoxel** [comment]
I have gotten around to implementing the create api on my website and the create api works and returns the expected data back.

**Comment by antoinecharbonneau** [comment]
Looking forward to use this crate when this is implemented, do let me know if there is something I can help with

**Comment by antoinecharbonneau** [comment]
After some testing, I've noticed an issue in the deserialization of `CreatePaymentLinkResponse` where the API will respond with a `OrderFullfilmentType` of `DIGITAL`. It isn't anywhere I have found in the doc, except for in the example response of this page: https://developer.squareup.com/reference/square/checkout-api/create-payment-link

## Reproducing the error

Try to create a checkout URL using `CheckoutApi::create_payment_link` with a `QuickPay` order

## Error in rust

Error: Error deserializing: error decoding response body: unknown variant `DIGITAL`, expected one of `PICKUP`, `SHIPMENT`, `DELIVERY` at line 1 column 1597

## Proposed fix

A simple fix would be to add `Digital` to `OrderFullfilmentType` enum, though it is kind of in a nebulous state in regards to the official support and documentation. I also don't know the current state of the implementation of Digital fulfillment type within the Orders API as this is the latest update I have seen on the matter: https://developer.squareup.com/forums/t/orders-api-with-digital-fulfillment/2699

Since you are the maintainer of this library and there is a discrepancy between Square's doc and its functionalities, how do you want to proceed with this case @kcable194?

Edit: Added information about a way to reproduce the issue.

**Comment by kcable194** [comment]
> After some testing, I've noticed an issue in the deserialization of `CreatePaymentLinkResponse` where the API will respond with a `OrderFullfilmentType` of `DIGITAL`. It isn't anywhere I have found in the doc, except for in the example response of this page: https://developer.squareup.com/reference/square/checkout-api/create-payment-link
> ## Reproducing the error
> 
> Try to create a checkout URL using `CheckoutApi::create_payment_link` with a `QuickPay` order
> ## Error in rust
> 
> Error: Error deserializing: error decoding response body: unknown variant `DIGITAL`, expected one of `PICKUP`, `SHIPMENT`, `DELIVERY` at line 1 column 1597
> ## Proposed fix
> 
> A simple fix would be to add `Digital` to `OrderFullfilmentType` enum, though it is kind of in a nebulous state in regards to the official support and documentation. I also don't know the current state of the implementation of Digital fulfillment type within the Orders API as this is the latest update I have seen on the matter: https://developer.squareup.com/forums/t/orders-api-with-digital-fulfillment/2699
> 
> Since you are the maintainer of this library and there is a discrepancy between Square's doc and its functionalities, how do you want to proceed with this case @kcable194?
> 
> Edit: Added information about a way to reproduce the issue.

Great catch, thanks for bringing it to my attention. Let's go ahead and add Digital to the OrderFullfillmentType since it is a valid enum variant that can be returned from Square. I'm going to make some progress on the rest of the Checkout API today.

**Comment by kcable194** [comment]
Added Digital variant to branch: feature/checkout-api-implementation

**Comment by kcable194** [comment]
Checkout api and webhooks implemented in v2.5.0

**Pull request closed without merging**